src/pepr/operator/controllers/exemptions/exemptions.ts

```diff
@@ -1,160 +1,114 @@
 import { Log } from "pepr";
-import { policies } from "../../../policies/index";
-import { ExemptionElement, Matcher, Policy, UDSExemption } from "../../crd";
-
-type StoredMatcher = Matcher & { owner: string };
-type PolicyMap = Map<Policy, StoredMatcher[]>;
+import { PolicyMap, StoredMatcher } from "../../../policies";
+import { Policy, UDSExemption } from "../../crd";
+
+export enum WatchPhase {
+  Added = "ADDED",
+  Modified = "MODIFIED",
+  Deleted = "DELETED",
+  Bookmark = "BOOKMARK",
+  Error = "ERROR",
+}
 
 const isSame = (a: StoredMatcher, b: StoredMatcher) => {
   return (
     a.name === b.name && a.namespace === b.namespace && a.kind == b.kind && a.owner === b.owner
   );
 };
 
-function addIfIncludesPolicy(
-  policy: Policy,
-  policyMap: PolicyMap,
-  exemptionEl: ExemptionElement,
-  ownerId: string,
-) {
-  const storedMatchers = policyMap.get(policy) ?? [];
-  const matcherToStore = {
-    ...exemptionEl.matcher,
-    owner: ownerId,
-  };
-  const isDuplicate = storedMatchers.some(sm => isSame(sm, matcherToStore));
-
-  // add if not already added to policy's exemption list
-  if (exemptionEl.policies.includes(policy) && !isDuplicate) {
-    policyMap.set(policy, [...storedMatchers, matcherToStore]);
-  }
-}
-
-// Delete a matcher from a policy if the policy has been removed from its policy list
-function deleteIfPolicyRemoved(
-  policy: Policy,
-  policyMap: PolicyMap,
-  exemptionEl: ExemptionElement,
-  ownerId: string,
-) {
-  const matcher = {
-    ...exemptionEl.matcher,
-    owner: ownerId,
-  };
-  const storedMatchers = policyMap.get(policy) ?? [];
-
-  if (storedMatchers.some(sm => isSame(sm, matcher)) && !exemptionEl.policies.includes(policy)) {
-    policyMap.set(
-      policy,
-      storedMatchers.filter(sm => {
-        if (!isSame(sm, matcher)) return sm;
-      }),
-    );
-    Log.debug(`Removing ${matcher.name} from ${policy}`);
+// Iterate through each exemption block of CR and add matchers to PolicyMap
+function addToMap(map: PolicyMap, exemption: UDSExemption, log: boolean = true) {
+  const exemptions = exemption.spec?.exemptions ?? [];
+  for (const e of exemptions) {
+    const matcherToStore = {
+      ...e.matcher,
+      owner: exemption.metadata?.uid || "",
+    };
+
+    const policies = e.policies ?? [];
+    for (const p of policies) {
+      const storedMatchers = map.get(p) ?? [];
+      if (log) {
+        console.log(`Adding to ${p}: ${JSON.stringify([...storedMatchers, matcherToStore])}`);
```

Comment:
Still working my way through the rest of the PR, but one comment upfront - anywhere we are using `console.log` I think we should really be using `Log.debug` or `Log.info` depending on the specific case. This will ensure we have consistent logging and are using the same log method as all the other parts of the operator + admission pieces.

I'm also on the fence on the conditional for logging here. I think I see how it is used (only logging adds, not modifies?), but would prefer that we err on the side of more logging rather than less.

Reply:
good point about the logger. I forgot about `Log` and just defaulted to `console`. I added the boolean to that we don't log adding to the temporary map. the updates will still get logged, just from the `addToMap`.

Reply:
Ah okay, tracking. Sounds good.